### PR TITLE
Order addresses by alias on checkout

### LINKED ISSUE #21437: Address selectors reversed after edition in FO's checkout process
#### Describe the bug

In **FO > Checkout > Address selection**, if you have more than one address, for example 2 addresses, editing one address will change the order of the address selector.

#### Expected behavior

Address selectors should not be reversed.

#### Steps to Reproduce

1. Make sure you have 2 addresses or more for your customer account.
2. Add a product to your cart, and proceed to checkout
3. At the address selection step, edit the first address, push the form, you'll see that the addresses selectors are reversed.

#### Screenshots

Before editing address "Home"
Capture d’écran 2020-10-14 à 11 16 47
After editing address "Home"
Capture d’écran 2020-10-14 à 11 22 07

#### Additional information

* PrestaShop version: 1.7.6, 1.7.7, 1.7.8

### PR DESCRIPTION
**Pull request by tegbessou**
Add an order by in the getSimpleAddressSql method, to render addresses ordered by alias

| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Versions: 1.7.6, 1.7.7, 1.7.8. <br> Addresses send to checkout step address was not ordered by aliases
| Type?             | bug fix
| Category?         | FO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #21437.
| How to test?      | Follow steps from #21437: Make sure you have 2 addresses or more for your customer account. <br> Make an order with the first address. <br> Then do another order, but at the address selection step, edit the first address (used in the last order), push the form, you'll see that the addresses selectors are not reversed. <br>
| Possible impacts? | Addresses will also be ordered in Address listing in My Account

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/26200)

### CODE REVIEW

**Comment by matks** [comment]
Bravo @tegbessou for this very first contribution to the PrestaShop open source project!

File: classes/Customer.php [hunk 1]
```diff
@@ -790,7 +790,8 @@ public function getSimpleAddressSql($idAddress = null, $idLang = null)
                         `id_lang` = ' . (int) $idLang . '
                         AND `id_customer` = ' . (int) $this->id . '
                         AND a.`deleted` = 0
-                        AND a.`active` = 1';
+                        AND a.`active` = 1
+                    ORDER BY a.`alias`';
```
Comment (matks) [inline classes/Customer.php:794]: Did you verify whether `getSimpleAddressSql()` is used in another function / usecase, to make sure this change does not introduce and unwanted side-effect somewhere else? 😄

Comment (matks) [inline classes/Customer.php:794]: Maybe we should keep it lined up like the above statements? 😄 
```suggestion
                        ORDER BY a.`alias`';
```
this makes it a little easier to read

**Review by matks — COMMENTED** [review]
Logic seems good, I provided 1 feedback about codestyle and I would like a little check to be sure it's safe 😉

File: classes/Customer.php (hunk 1 above)
Comment (tegbessou) [reply classes/Customer.php:794]: The previous statement is 'AND', so I choose to align my 'ORDER BY' statement with the 'WHERE' statement. Are you sure I have to fix this?

Comment (tegbessou) [reply classes/Customer.php:794]: The `getSimpleAddressSQL()` is used in another function like AddressController, so this order also address in AddressController on FO. I think it's a good thing, like this we are consistent.

What do you think about ?

Comment (matks) [reply classes/Customer.php:794]: I just want to be sure that the changes we will bring in AddressController are something we want. I agree it seems better to always order adresses by alias.

Can you add this side-effect in the body of your PR (last line "Possible impacts" to inform QA team of this piece of information) ?

Comment (matks) [reply classes/Customer.php:794]: Oh! I did not open the PR wide so I did not see the WHERE was not lined up.

You're right this is better this way 👍

**Review by matks — DISMISSED** [review]
Thank you for answers, I approve the Pull Request.

File: classes/Customer.php (hunk 1 above)
Comment (tegbessou) [reply classes/Customer.php:794]: I have add this in the "Possible impacts" section

Comment (matks) [reply classes/Customer.php:794]: Perfect! this will help QA team to validate your contribution

**Comment by tegbessou** [comment]
Thank for your review @matks

**Review by PierreRambaud — APPROVED** [review]

**Review by matks — APPROVED** [review]

**Comment by florine2623** [comment]
Hello @tegbessou ,

Congrats on your first contribution and thanks for the PR !
It is QA tested and QA approved !

**Comment by tegbessou** [comment]
@florine2623 thank you for the validation :)

**Comment by Progi1984** [comment]
Thank you @tegbessou & @florine2623

**Pull request merged**